src/Presentation/SofortUrlGenerator.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\Presentation;
+
+use Sofort\SofortLib\Sofortueberweisung;
+use WMDE\Euro\Euro;
+use RuntimeException;
+
+class SofortUrlGenerator {
+
+	private const CURRENCY = 'EUR';
+
+	/**
+	 * @var \WMDE\Fundraising\Frontend\Presentation\SofortUrlConfig
+	 */
+	private $config;
+	/**
+	 * @var \Sofort\SofortLib\Sofortueberweisung
+	 */
+	private $api;
+
+	public function __construct( SofortUrlConfig $config, Sofortueberweisung $api ) {
+		$this->config = $config;
+		$this->api = $api;
+	}
+
+	public function generateUrl( string $itemId, Euro $amount, string $accessToken ): string {
```

Comment:
Chances are, we only need one id to id a donation. As this offers us an interface to pass an identifying criteria into the generator it should be fine. The using side may need a change, [as you correctly indicated there](https://github.com/wmde/FundraisingFrontend/pull/910#discussion_r124843683) (i.e. use of a different donation method to generate an _id_)